### PR TITLE
CLOUDP-178273 : [Monitoring and Logs API] Migrate DownloadLog Command to v2 SDK

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_  [CLOUDP-178273](https://jira.mongodb.org/browse/CLOUDP-178273)

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/logs.go [hunk 1]
```diff
@@ -82,7 +85,26 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
 func (s *Store) DownloadLog(groupID, host, name string, out io.Writer, opts *atlas.DateRangetOptions) error {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
-		_, err := s.client.(*atlas.Client).Logs.Get(s.ctx, groupID, host, name, out, opts)
+		fileBaseName := strings.TrimSuffix(name, filepath.Ext(name))
+		result := s.clientv2.MonitoringAndLogsApi.GetHostLogs(s.ctx, groupID, host, fileBaseName)
+		if opts != nil {
+			if opts.StartDate != "" {
```
Comment (drinkbird) [inline internal/store/logs.go:91]: If this comes as `nil` instead of an empty string, what will the code do?

File: internal/store/logs.go [hunk 2]
```diff
@@ -82,7 +85,26 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
...
+				start, _ := strconv.ParseInt(opts.StartDate, 10, 64)
+				result = result.StartDate(start)
```
Comment (drinkbird) [inline internal/store/logs.go:93]: if the parse operation above fails, what will be assigned here? nil or empty string? 

if opts.EndDate is nil, what will the code do?

File: internal/store/logs.go (hunk 1 above)
Comment (wtrocki) [reply internal/store/logs.go:91]: In this particular example we dealing with string datatype that cannot be nil. 
https://github.com/mongodb/go-client-mongodb-atlas/blob/c3435ec31149df5973ef01f992bb72d598615a39/mongodbatlas/logs.go#L41-L42

File: internal/store/logs.go [hunk 3]
```diff
@@ -82,7 +85,26 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
 func (s *Store) DownloadLog(groupID, host, name string, out io.Writer, opts *atlas.DateRangetOptions) error {
```
Comment (wtrocki) [inline internal/store/logs.go:85]: We have dedicated v2 objects to organize those arguments. Can we use that?

File: internal/store/logs.go (hunk 1 above)
Comment (wtrocki) [reply internal/store/logs.go:91]: The problem is that we should not use v1 objects itself and v2 will actually have pointer

Comment (swattyT) [reply internal/store/logs.go:91]: `opts.StartDate` is declared as string 
```
type DateRangetOptions struct {
    StartDate string `url:"startDate,omitempty"`
    EndDate   string `url:"endDate,omitempty"`
}
```
ParseInt only accepts string value in the argument.

File: internal/store/logs.go [hunk 4]
```diff
@@ -82,7 +85,26 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
...
+				end, _ := strconv.ParseInt(opts.EndDate, 10, 64)
+				result = result.EndDate(end)
+			}
+		}
+		logs, _, err := result.Execute()
+		if err != nil {
+			return err
+		}
+		if logs == nil {
+			return fmt.Errorf("returned file is empty")
```
[the remark this answers is not in the document]
Comment (swattyT) [reply internal/store/logs.go:105]: This is to return an `error` object.

File: internal/store/logs.go (hunk 3 above)
Comment (swattyT) [reply internal/store/logs.go:85]: Sure, I will use `GetHostLogsWithParams` instead.

**Comment by wtrocki** [comment]
[nit] [optional] I think this PR might be using the old approach for fetching data from the backend. 
With a number of arguments and parseInt required I would consider simplifying using `WithParams` struct.
See: https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_1_concepts.md#fetching-data-from-the-back-end

Otherwise looks good

**Review by wtrocki — APPROVED** [review]
LGTM. Thank you for using v2 models.

**Review by matt-condon — APPROVED** [review]

File: internal/store/logs.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (drinkbird) [reply internal/store/logs.go:90]: I was under the impression that there is an `fmt.Error(` (without the `f`) but there isn't one, so all good

[the remark this answers is not in the document]
Comment (wtrocki) [reply internal/store/logs.go:90]: [errors.New()](https://pkg.go.dev/errors#New) enters the chat

File: internal/store/logs.go (hunk 2 above)
Comment (wtrocki) [reply internal/store/logs.go]: @swattyT I would recommend logging improvement jira to cover the date time for the atlas CLI and SDKs.
Both systems have different standards but are always the same so we can write familiar helper and do transformation directly at the Cobra flag level. That will let us leave complex `start, _ := strconv.ParseInt(opts.StartDate, 10, 64)` out of the future code reviews.

**Pull request merged**